src/applications/yellow-ribbon/containers/SearchForm/index.jsx

```diff
@@ -24,40 +25,101 @@ export class SearchForm extends Component {
 
     // Derive the state values from our query params.
     const city = queryParams.get('city') || '';
+    const contributionAmount = queryParams.get('contributionAmount') || '';
     const country = queryParams.get('country') || '';
     const name = queryParams.get('name') || '';
+    const numberOfStudents = queryParams.get('numberOfStudents') || '';
     const state = queryParams.get('state') || '';
 
     this.state = {
       city,
+      contributionAmount,
       country,
       name,
+      numberOfStudents,
       state,
     };
   }
 
   componentDidMount() {
-    const { country, city, name, state } = this.state;
+    const {
+      city,
+      contributionAmount,
+      country,
+      name,
+      numberOfStudents,
+      state,
+    } = this.state;
 
     // Fetch the results with their name if it's on the URL.
-    if (city || country || name || state) {
-      this.props.fetchResultsThunk({ city, country, name, state });
+    if (
+      city ||
+      contributionAmount ||
+      country ||
+      name ||
+      numberOfStudents ||
+      state
+    ) {
+      this.props.fetchResultsThunk({
+        city,
+        contributionAmount,
+        country,
+        name,
+        numberOfStudents,
+        state,
+      });
     }
   }
 
+  onCheckboxChange = key => () => {
+    // Uncheck the checkbox.
+    if (this.state[key]) {
+      this.setState({ [key]: '' });
+      return;
+    }
+
+    // Check the checkbox.
+    this.setState({ [key]: 'unlimited' });
```

Comment:
Hm, it would be cool if instead of a floating string value we could use a constant, or tie it closer to the API request since it's kind of like an enum value to the API method.

Reply:
Hmm... so like create a constant string in our `constants` folder?

Reply:
It's probably fine here! Would really only be necessary if it was floating all over the place